packages/front-end/components/Settings/UpgradeModal/index.tsx

```diff
@@ -80,37 +78,46 @@ export default function UpgradeModal({
 
   const { organization, refreshOrganization } = useUser();
 
+  function getLowestAvailablePlan() {
+    // If no commercialFeature is passed in, the lowestAvailablePlan is the plan higher than their effectiveAccountPlan
+    if (!commercialFeature) {
+      if (
+        !effectiveAccountPlan ||
+        ["oss", "starter"].includes(effectiveAccountPlan)
+      ) {
+        return "pro";
+      }
+
+      if (
+        effectiveAccountPlan &&
+        ["pro", "pro_sso"].includes(effectiveAccountPlan)
+      ) {
+        return "enterprise";
+      }
+    } else {
+      return commercialFeatureLowestPlan?.[commercialFeature] || "starter";
+    }
+  }
+
```

Comment:
Previously, if no `commercialFeature` was passed in, it would be possible for an `Enterprise` org to get the `Pro` upgrade modal.

Now, if no `commercialFeature` is passed in, the `lowestAvailablePlan` is the plan ` higher than their current plan.